pkg/controller/backup-operator/sync.go

```diff
@@ -127,24 +251,35 @@ func (b *Backup) handleBackup(spec *api.BackupSpec) (*api.BackupStatus, error) {
 	if spec.BackupPolicy != nil && spec.BackupPolicy.TimeoutInSecond > 0 {
 		backupTimeout = time.Duration(spec.BackupPolicy.TimeoutInSecond) * time.Second
 	}
+	backupMaxCount := 0
+	if spec.BackupPolicy != nil && spec.BackupPolicy.MaxBackups > 0 {
+		backupMaxCount = spec.BackupPolicy.MaxBackups
+	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), backupTimeout)
+	if parentContext == nil {
+		tmpParent := context.Background()
+		parentContext = &tmpParent
+	}
+	ctx, cancel := context.WithTimeout(*parentContext, backupTimeout)
 	defer cancel()
 	switch spec.StorageType {
 	case api.BackupStorageTypeS3:
-		bs, err := handleS3(ctx, b.kubecli, spec.S3, spec.EtcdEndpoints, spec.ClientTLSSecret, b.namespace)
+		bs, err := handleS3(ctx, b.kubecli, spec.S3, spec.EtcdEndpoints, spec.ClientTLSSecret,
+			b.namespace, isPeriodicBackup(spec), backupMaxCount)
 		if err != nil {
```

Comment:
can't we pass the value to a variable for isPeriodicBackup(spec) instead of checking each time?